ibis/backends/tests/test_register.py

```diff
@@ -287,10 +295,12 @@ def test_register_pandas(con):
     pd = pytest.importorskip("pandas")
     df = pd.DataFrame({"x": [1, 2, 3], "y": ["a", "b", "c"]})
 
-    t = con.register(df)
+    with pytest.warns(FutureWarning, match="v9.0"):
+        t = con.register(df)
```

Comment:
The in-memory tests we can probably leave alone for now, but we should add a comment that these should be updated  to work with the upcoming `read_in_memory` and we should also add a polars dataframe to the mix